node/actors/os_procs.go

```diff
@@ -60,23 +61,35 @@ func (proc *OsProcess) Run() error {
 		}
 		return err
 	}
+
+	proc.logger.Debug("OS process terminated successfully", slog.Int("pid", proc.cmd.Process.Pid))
+	proc.cmd = nil
+
+	return nil
+}
+
+// Sends an interrupt signal to the running process (if it's running)
+func (proc *OsProcess) Stop(reason string) error {
```

Comment:
Do we want to call this interrupt instead of `Stop` ? Is there a scenario where interrupt doesn't stop a process?